src/features/security-cloud-game/SecurityCloudGame.js

```diff
@@ -86,156 +112,130 @@ class SecurityCloudGame extends Component {
-
-        const dragRight = dragLeft + cloudSize;
-        const dragBottom = dragTop + cloudSize;
-
-        const { x: dropLeft, y: dropTop } = dropCloudPosition;
-        const dropMidX = dropLeft + cloudSize / 2;
-        const dropMidY = dropTop + cloudSize / 2;
-
-        return !(dragBottom < dropMidY || dragTop > dropMidY || dragLeft > dropMidX || dragRight < dropMidX);
-    }
+    const getAccessibilityInstructions = () =>
+        gameBoardHeight &&
+        cloudSize &&
+        gridTrackSize &&
+        formatMessage(
+            messages.accessibilityInstructions,
+            getGridPosition({ x: width - cloudSize, y: gameBoardHeight - cloudSize }, gridTrackSize),
+        );
 
     /**
      * Renders the drop cloud.
      * @returns {JSX}
      */
-    renderDropCloud() {
-        const { isValidDrop, dropCloudPosition } = this.state;
-
+    const renderDropCloud = () => {
         if (dropCloudPosition && !isValidDrop) {
-            const { cloudSize } = this.props;
-
-            // return the drop region with a DragCloud and DropCloud by default
             return (
-                <DropCloud
-                    className={this.state.isOverlap ? 'is-over' : ''}
-                    cloudSize={cloudSize}
-                    position={dropCloudPosition}
-                />
+                <DropCloud className={isOverlap ? 'is-over' : ''} cloudSize={cloudSize} position={dropCloudPosition} />
             );
         }
 
         return null;
-    }
+    };
 
     /**
      * Renders the drag cloud.
      * @returns {JSX}
      */
-    renderDragCloud() {
-        const { isValidDrop, dragCloudPosition } = this.state;
+    const renderDragCloud = () => {
+        const { current: gameBoardSize } = gameBoardSizeRef;
 
         if (dragCloudPosition) {
             return (
-                <Draggable bounds="parent" disabled={isValidDrop} onDrag={this.onDrag} onStop={this.onDragStop}>
-                    <DragCloud cloudSize={this.props.cloudSize} position={this.state.dragCloudPosition} />
-                </Draggable>
+                <DragCloud
+                    gameBoardSize={gameBoardSize}
+                    cloudSize={cloudSize}
+                    disabled={isValidDrop}
+                    gridTrackSize={gridTrackSize}
+                    onDrop={onDrop}
+                    position={dragCloudPosition}
+                    updateLiveText={updateLiveText}
+                    updatePosition={updatePosition}
+                />
             );
         }
 
         return null;
-    }
+    };
 
     /**
      * Renders the message shown to the user
      * @returns {JSX}
      */
-    renderMessage() {
-        if (this.state.isValidDrop) {
+    const renderMessage = () => {
+        if (isValidDrop) {
             return <FormattedMessage {...messages.success} />;
         }
 
         return <FormattedMessage {...messages.instructions} />;
-    }
+    };
+
+    const renderAriaLiveText = () => (
```

Comment:
this component seems small. thoughts on adding it directly into the return on line 208?